pipeline-templates/split-combine/splitcombine.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+    "bufio"
+    "flag"
+    "fmt"
+    "log"
+    "os"
+    "sort"
+    "path/filepath"
+)
+
+func main() {
+    mode := flag.String("mode", "split", "Specify mode: split or combine")
+    input := flag.String("input", "", "Specify input file or folder")
+    output := flag.String("output", "", "Specify output folder or file")
+    lineCount := flag.Int("linecount", 0, "Specify number of lines per file in split mode")
+
+    flag.Parse()
+
+    if *mode == "split" {
+        if *lineCount == 0 {
+            log.Fatal("Line count must be provided in split mode")
+        }
+        if *input == "" {
+            log.Fatal("Input file must be provided in split mode")
+        }
+        if *output == "" {
+            log.Fatal("Output folder must be provided in split mode")
+        }
+
+        err := walkDirToSplit(*input, *output, *lineCount)
+        if err != nil {
+            log.Fatal(err)
+        }
+    } else if *mode == "combine" {
+        if *input == "" {
+            log.Fatal("Input folder must be provided in combine mode")
+        }
+        if *output == "" {
+            log.Fatal("Output file must be provided in combine mode")
+        }
+
+        err := combineFiles(*input, *output)
+        if err != nil {
+            log.Fatal(err)
+        }
+    } else {
+        log.Fatal("Invalid mode specified. Must be split or combine")
+    }
+
+}
+
+func splitFileNameAndExt(filePath string) (string, string) {
+    fileExt := filepath.Ext(filePath)
+    fileName := filepath.Base(filePath[:len(filePath)-len(fileExt)])
+    return fileName, fileExt
+}
+
+func walkDirToSplit(path string, outputFolder string, lineCount int) error {
+        files, err := getFileListDir(path)
+        if err != nil {
+            return err
+        }
+        for _, file := range files {
+            splitFile(file, outputFolder, lineCount)
+        }
+        return nil
+    }
+
+func getFileListDir(path string)([]string, error) {
+    var files []string
+    err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+        if err != nil {
+            return err
+        }
+        if !info.IsDir() {
+            files = append(files, path)
+        }
+        return nil
+    })
+    if err != nil {
+        return nil, err
+    }
+    sort.Strings(files)
+    return files, nil
+}
+
+
+func splitFile(inputFile string, outputFolder string, lineCount int) error {
```

Comment:
Is this only for `csv` files, or will this handle `txt`, `rtf`, and `tex`, for example? And does it only split on new lines?

Reply:
any text format. splits on newlines only. anything else you'd like to see?

Reply:
Splitting on `,` or ` ` could be nice but not completely necessary. No concrete use cases come to mind though.